### PR TITLE
Feature/downvote

### PR DESCRIPTION
**Pull request by mistikk**

### CODE REVIEW

File: src/components/upvote/container/upvoteContainer.js [hunk 1]
```diff
@@ -53,7 +53,12 @@ class UpvoteContainer extends PureComponent {
     } = this.props;
 
     const author = get(content, 'author');
-    const isVoted = get(content, 'is_voted');
+    const isVoted = get(content, 'is_voted')
```
Comment (ue) [inline src/components/upvote/container/upvoteContainer.js]: in short case;

`const isVoted = get(content, 'is_voted', false)
      && parseInt(get(content, 'is_voted'), 10) / 10000;`

File: src/components/upvote/container/upvoteContainer.js [hunk 2]
```diff
@@ -53,7 +53,12 @@ class UpvoteContainer extends PureComponent {
...
+      ? parseInt(get(content, 'is_voted'), 10) / 10000
+      : false;
+    const isDownVoted = get(content, 'is_down_voted')
```
Comment (ue) [inline src/components/upvote/container/upvoteContainer.js]: both

File: src/components/upvote/view/upvoteView.js [hunk 3]
```diff
@@ -156,9 +221,10 @@ class UpvoteView extends Component {
       iconType = 'AntDesign';
     }
 
-    const _percent = `${(sliderValue * 100).toFixed(0)}%`;
+    const _percent = `${downvote ? '-' : ''}${(sliderValue * 100).toFixed(0)}%`;
     const _amount = `$${amount}`;
     const _totalPayout = totalPayout || '0.000';
+    const _sliderColor = downvote ? '#ec8b88' : '#357ce6';
```
Comment (ue) [inline src/components/upvote/view/upvoteView.js]: there is no `sliderColor` underscore not necessary

File: src/components/upvote/view/upvoteView.js [hunk 4]
```diff
@@ -183,6 +249,14 @@ class UpvoteView extends Component {
                       isShow={!isVoting}
                     />
                   </View>
+                ) : isDownVoted ? (
+                  <Icon
```
Comment (ue) [inline src/components/upvote/view/upvoteView.js]: can merge with follow icon like inside the props can use is down voted

Comment (ue) [reply src/components/upvote/view/upvoteView.js]: write to me from dm if the comment not explainable

File: src/utils/postParser.js [hunk 5]
```diff
@@ -144,5 +148,22 @@ export const parseComments = async (comments, currentUserName) => {
...
+  );
+  if (result) {
+    return result.percent;
+  }
+  return false;
+};
+
+const isDownVoted = (activeVotes, currentUserName) => {
+  const result = activeVotes.find(
+    element => element.voter === currentUserName && element.percent < 0,
```
Comment (ue) [inline src/utils/postParser.js]: can use `get`

**Comment by ue** [comment]
@mistikk when pressing the downvote button then changing slide the price value changing I don't thing so if its necessary am I right @feruzm ?

File: src/components/upvote/container/upvoteContainer.js (hunk 1 above)
Comment (mistikk) [reply src/components/upvote/container/upvoteContainer.js]: this can be wrong line?

File: src/components/upvote/view/upvoteView.js (hunk 3 above)
Comment (feruzm) [reply src/components/upvote/view/upvoteView.js]: On video Mustafa post, I saw the color, maybe on mobile it is not visible?!

**Comment by feruzm** [comment]
@ue we want user to push downvote button twice, that's why different value/slider on first click and casting downvote on second click is good way.

https://cdn.discordapp.com/attachments/487610005558657034/613810302227709973/Screen_Recording_2019-08-21_at_21.52.57.mov

**Comment by mistikk** [comment]
@ue updated

File: src/utils/postParser.js [hunk 6]
```diff
@@ -144,5 +148,22 @@ export const parseComments = async (comments, currentUserName) => {
   return _comments;
 };
 
-const isVoted = (activeVotes, currentUserName) =>
-  activeVotes.some(v => v.voter === currentUserName && v.percent > 0);
+const isVoted = (activeVotes, currentUserName) => {
+  const result = activeVotes.find(
+    element => element.voter === currentUserName && element.percent > 0,
```
Comment (ue) [inline src/utils/postParser.js]:
```suggestion
    element => get(element, 'voter') === currentUserName && get(element, 'percent', 0) > 0,
```

**Pull request merged**